ocf_data_sampler/load/gsp.py

```diff
@@ -41,11 +45,18 @@ def open_gsp(zarr_path: str, boundaries_version: str = "20220314") -> xr.DataArr
     df_gsp_loc = get_gsp_boundaries(boundaries_version)
 
     # Open the GSP generation data
-    ds = (
+    if public:
```

Comment:
I wond if its slightly neater is you sya
```
backend_kwargs = {}
if public:
   backend_kwargs={"storage_options":{"anon": True}}
```
and then add `backend_kwargs` to `open_dataset`